src/main/java/com/minecolonies/coremod/commands/ActionArgumentType.java

```diff
@@ -0,0 +1,360 @@
+                            }
+                        }
+                        return middleNameSet.stream().filter(k -> k.startsWith(potentiaCitizenNameParts[1])).collect(Collectors.toList());
+                    }
+                    else
+                    {
+                        final Set<String> lastNameSet = new HashSet<>();
+                        for (final String citizenName : citizenNameStrings)
+                        {
+                            final String[] citizenNameParts = citizenName.split(" ");
+                            if ((potentiaCitizenNameParts[0].equals(citizenNameParts[0])) && (potentiaCitizenNameParts[1].equals(citizenNameParts[1])))
+                            {
+                                lastNameSet.add(citizenNameParts[2]);
+                            }
+                        }
+                        return lastNameSet.stream().filter(k -> k.startsWith(potentiaCitizenNameParts[2])).collect(Collectors.toList());
+                   }
+                }
+                break;
+        }
+
+        return Collections.emptyList();
+    }
+
+    public Object parse(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @Nullable final BlockPos pos,
+            @NotNull final List<ActionMenuHolder> parsedHolders,
+            final String potentialArgumentValue)
+    {
+        // TODO: selector support, such as used by CommandKill to find player
+        // Entity entity = <net.minecraft.command.CommandBase>.getEntity(server, sender, args[0]);
+
+        switch (this)
+        {
+            case INTEGER:
+                try
+                {
+                    return Integer.parseInt(potentialArgumentValue);
+                }
+                catch (final NumberFormatException e)
+                {
+                    return null;
+                }
+            case BOOLEAN:
+                return Boolean.parseBoolean(potentialArgumentValue);
+            case COORDINATE_X:
+            case COORDINATE_Y:
+            case COORDINATE_Z:
+                try
+                {
+                    return Integer.parseInt(potentialArgumentValue);
+                }
+                catch (final NumberFormatException e)
+                {
+                    return null;
+                }
+            case PLAYER:
+                final List<String> playerNameStrings = getOnlinePlayerNames(server);
+                if (playerNameStrings.contains(potentialArgumentValue))
+                {
+                    return server.getPlayerList().getPlayerByUsername(potentialArgumentValue);
+                }
+                else
+                {
+                    if ("[abandoned]".equals(potentialArgumentValue))
+                    {
+                        return new FakePlayer(server.getWorld(0), new GameProfile(UUID.randomUUID(), "[abandoned]"));
+                    }
+                    return null;
+                }
+            case COLONY:
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') End this switch case with an unconditional break, return or throw statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS128)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this switch case number of lines from 24 to at most 10, for example by extracting code into methods. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1151)